### PR TITLE
fix(example): update example and fix bugs in Aqua code

### PR DESCRIPTION
**Pull request by justprosh**
- closes NET-190
- rename neighbour -> neighbor
- fix issue in aqua code

### CODE REVIEW

**Comment by boneyard93501** [comment]
committing fluence.yaml poses a problem to people following along:
```bash
fluence service add echoService
 ›   Warning: You already have echoService in fluence.yaml
? Enter another name for the service (must start with a lowercase letter and contain only letters, numbers, and underscores)
```
the choices are to not commit any of the fluence cli files and folders or to tell users wanting to follow along to delete them.

in the readme, this `If you have Apple Silicon you should install cargo ` should be `If you have Apple Silicon you should install marine `

also, you should tell users to `npm i` before ` fluence run -f 'echo("hi")'` (not after).

`registerResource()` doesn't seem to work:

```bash
 fluence run -f 'registerResource()'
Running:
  function: registerResource()
  relay: /dns4/kras-06.fluence.dev/tcp/19001/wss/p2p/12D3KooWDUszU2NeWyUVjCXhGEt1MoZrhvdmaQQwtZUriuGN1jTr
... done

Result:

[
  "Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur"
]

mbp16~/localdev/registry/example(fix_example|…) % fluence run -f 'registerService("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur")
quote> '
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
  function: registerService("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur")

  relay: /dns4/kras-05.fluence.dev/tcp/19001/wss/p2p/12D3KooWCMr9mU894i8JXAFqpgoFtx6qnV1LFPSfVc3Y34N4h4LS
... done

Result:

[
  [
    false,
    false
  ]
]
``` 

step 7: i get different output:

```bash
fluence run -f 'echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")'
Running:
  function: echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")
  relay: /dns4/kras-09.fluence.dev/tcp/19001/wss/p2p/12D3KooWD7CvsYcpF9HE9CCV9aY3SJ317tkXVykjtZnht2EbzDPm
... done

Result:

"12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: msg"
```

also different output:

```bash
mbp16~/localdev/registry/example(fix_example|…) % fluence run -f 'echoAll("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur", "hi")'
Running:
  function: echoAll("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur", "hi")
  relay: /dns4/kras-08.fluence.dev/tcp/19001/wss/p2p/12D3KooWFtf3rfCDAfWwt6oLZYZbDfn9Vn7bv7g6QjjQxUUEFVBt
... done

Result:

[
  [
    "12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: hi"
  ]
]

```

and same for `unregisterEchoService`:

```bash
fluence run -f 'echoAll("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur", "hi")'
Running:
  function: echoAll("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur", "hi")
  relay: /dns4/kras-08.fluence.dev/tcp/19001/wss/p2p/12D3KooWFtf3rfCDAfWwt6oLZYZbDfn9Vn7bv7g6QjjQxUUEFVBt
... done

Result:

[
  [
    "12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: hi"
  ]
]

mbp16~/localdev/registry/example(fix_example|…) % fluence run -f 'unregisterEchoService("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur")
'
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
Running:
  function: unregisterEchoService("Hr2sAZww931uG5H6AH9nqRiXQRpmFgxAzoWLxeZA5tur")

  relay: /dns4/kras-05.fluence.dev/tcp/19001/wss/p2p/12D3KooWCMr9mU894i8JXAFqpgoFtx6qnV1LFPSfVc3Y34N4h4LS
... done

Result:

[
  [
    false,
    false
  ]
]
```
and of course re-running the service works.

File: example/README.md [hunk 1]
```diff
@@ -2,27 +2,28 @@
 
 ## Overview
 
-The example shows one of the important Registry use-cases — services advertisement and discovery. So it is about advertisement by providers, discovery and usage by clients with the same Aqua code transparently without any knowledge about particular peer and service ids.
+The example shows one of the important Registry use-cases — services advertisement and discovery. So it is about advertisement, discovery and usage by clients with the same Aqua code transparently without any knowledge about particular peer and service ids.
```
Comment (boneyard93501) [inline example/README.md:5]: that doesn't sound right. first, service addr params need to be known for registration and Aqua also uses them. so the argument, i guess is, that the aqua function developer doesn't need to explicitly specify service addrs if a registry record is available.

File: example/README.md [hunk 2]
```diff
@@ -2,27 +2,28 @@
...
 
 In the beginning, we will deploy a Rust echo service and call it with the Fluence CLI.
 Then we will start a JS/TS client with echo service and run it.
 
-Finally, we will use Registry to call different types of services with exactly one piece of code.
+Secondly, we will use Registry to call different types of services with exactly one piece of code.
```
Comment (boneyard93501) [inline example/README.md:10]: i only see the EchoService. what other services are you calling?

File: example/README.md [hunk 3]
```diff
@@ -2,27 +2,28 @@
...
+@fluencelabs/cli: 0.2.13
 ```
 
 # Set up environment
 
 1. Install `fluence` cli:
 
     ```bash
-    npm i -g @fluencelabs/cli@0.2.7
+    npm i -g @fluencelabs/cli@0.2.13
```
Comment (boneyard93501) [inline example/README.md:26]: that's risky. as we have seen with the bad version management upstream.

File: example/README.md [hunk 4]
```diff
@@ -73,7 +74,7 @@ rust: rustc 1.63.0-nightly
 
     ```bash
     # for m1:
-    cargo install mrepl --version '0.18.0' --root ~/.fluence/cargo
+    cargo install mrepl --version '0.18.6' --root ~/.fluence/cargo
     ```
```
Comment (boneyard93501) [inline example/README.md:78]: currently, that has no impact from a version perspective. whatever fluence cli specifies, it looks for. so there needs to be a way for the user to figure out what fluence cli wants before they manually install.

File: example/README.md [hunk 5]
```diff
@@ -283,36 +284,37 @@ Firstly, we need to create a **resource.** A resource represents a group of serv
...
+   So the resource id is `5pYpWB3ozi6fi1EjNs9X5kE156aA6iLECxTuVdJgUaLB`. Please note that the resource id might be different in your case.
 
-    Using the resource id we can access any registered provider of the resource. There can be more registered echo services on different peers that we can use transparently.
+    Using the resource id we can access any registered service. There can be more registered echo services on different peers that we can use transparently.
 
 3. This [code](src/aqua/main.aqua#L28) registers deployed service by given `resource_id`:
 
     ```rust
-    func registerService(resource_id: string) -> bool:
+    func registerService(resource_id: string) -> *bool:
```
Comment (boneyard93501) [inline example/README.md:298]: in `resources-api.aqua` shouldn't `unregisterService` be `unregisterResource` as the tombstone is over resource_id

**Review by boneyard93501 — CHANGES_REQUESTED** [review]

File: example/README.md (hunk 5 above)
Comment (justprosh) [reply example/README.md:298]: Nope, it's `unregisterService`, peer_id is passed implicitly

**Comment by justprosh** [comment]
> committing fluence.yaml poses a problem to people following along:
@boneyard93501 
fluence.yaml is not in the repo

File: example/README.md (hunk 2 above)
Comment (justprosh) [reply example/README.md:10]: Will rephase, I meant js and rust services

**Comment by justprosh** [comment]
>also, you should tell users to npm i before fluence run -f 'echo("hi")' (not after).

@boneyard93501 why?

**Comment by justprosh** [comment]
@boneyard93501 I checked the full example from scratch one more time, and everything works for me. Also, it seems that you already had fluence.yml in your local repo and you tried to deploy two services

File: example/README.md (hunk 3 above)
Comment (boneyard93501) [reply example/README.md:26]: maybe we should add somewhere in the readme that if any of the fluence cli commands fail due to some version mismatch to use the `fluence dependency` command to manually specify the correct version?

**Comment by boneyard93501** [comment]
looks good.

**Review by boneyard93501 — APPROVED** [review]

**Pull request merged**